src/index.js

```diff
@@ -25,146 +25,135 @@ import InstructorsOnly from './components/InstructorsOnly'
 import LoggedOut from './components/LoggedOut'
 import RouteNotFound from './components/RouteNotFound'
 
-class App extends Component {
+const App = () => {
 
-  state = {
-    userUrl: false,
-  }
+  const userUrl = login()
 
-  componentDidMount() {
-    this.setState({
-      userUrl: login(),
-    })
+  if(!userUrl) {
+    logout()
```

Comment:
This line will auto logout/clear token if user has cached old token without `user_url`